babel/numbers.py

```diff
@@ -774,6 +774,8 @@ def format_currency(
                                           locale=locale, currency_digits=currency_digits,
                                           decimal_quantization=decimal_quantization, group_separator=group_separator,
                                           numbering_system=numbering_system)
+    if locale is None:
+        locale = default_locale()
```

Comment:
```suggestion
        locale = LC_NUMERIC
```

Though I'm not sure if it makes sense to just fix this one function. A lot of the other `format_*` functions cannot deal with explicitly passed `None` as the locale either:

```python
from babel.numbers import format_compact_currency
format_compact_currency('12', 'EUR', locale=None)
```
I think if we're gonna fix this, we should fix all of them at the same time, otherwise we're gonna have functions with the same type annotation behaving differently which could lead to confusion.